lib/remoteproc/rsc_table_parser.c

```diff
@@ -146,9 +150,21 @@ int handle_vdev_rsc(struct remoteproc *rproc, void *rsc)
 						  notifyid + 1);
 		if (notifyid != RSC_NOTIFY_ID_ANY)
 			vring_rsc->notifyid = notifyid;
+		else
+			goto err;
 	}
 
 	return 0;
+
+err:
+
+	for (i--; i >= 0; i--) {
```

Comment:
infinite loop?
warning: comparison of unsigned expression >= 0 is always true [-Wtype-limits]
  161 |  for (i--; i >= 0; i--) {
      |              ^~

Reply:
Yes, sorry - I need to change i to int.